server/auth/store_test.go

```diff
@@ -450,6 +450,40 @@ func TestRoleGrantPermission(t *testing.T) {
 	}
 }
 
+func TestRootRoleGrantPermission(t *testing.T) {
+	as, tearDown := setupAuthStore(t)
+	defer tearDown(t)
+
+	perm := &authpb.Permission{
+		PermType: authpb.WRITE,
+		Key:      []byte("Keys"),
+		RangeEnd: []byte("RangeEnd"),
+	}
+	_, err := as.RoleGrantPermission(&pb.AuthRoleGrantPermissionRequest{
+		Name: "root",
+		Perm: perm,
+	})
+
+	if err != nil {
+		t.Error(err)
+	}
+
+	r, err := as.RoleGet(&pb.AuthRoleGetRequest{Role: "root"})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	//whatever grant permission to root, it always return root permission.
+	expectPerm := &authpb.Permission{
+		PermType: authpb.READWRITE,
+		RangeEnd: []byte{0},
+	}
+
+	if !reflect.DeepEqual(expectPerm, r.Perm[0]) {
```

Comment:
pls, prefer: https://github.com/stretchr/testify/assert : assert.Equal(...)

Reply:
Hi.I just found the same jugdement  at other place in this file.

Reply:
already modify the judgement way in this file.